core/src/main/scala/io/projectglow/vcf/BigVCFDatasource.scala

```diff
@@ -80,5 +92,7 @@ object BigVCFDatasource extends HlsUsageLogging {
 
         Iterator(baos.toByteArray)
     }
+    rdd.unpersist()
```

Comment:
Hm, I'm not sure that it makes sense to unpersist here. Then the RDD will be uncached when it is materialized again.